src/XMakeTasks/Hash.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.Build.Tasks
+{
+    /// <summary>
+    /// Generates a hash of a given ItemGroup items. Metadata is not considered in the hash.
+    /// <remarks>Currently uses SHA1. Implementation subject to change between MSBuild versions.</remarks>
```

Comment:
Maybe some sort of "Not intended as a cryptographic security measure" type caveat?